segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/ContainerTableExtensionImpl.java

```diff
@@ -457,13 +463,48 @@ protected int getMaxCompactionSize() {
                 });
     }
 
+    @SuppressWarnings("unchecked")
+    public <T> CompletableFuture<AsyncIterator<IteratorItem<T>>> newDeltaIterator(@NonNull String segmentName, long fromPosition, @NonNull Duration fetchTimeout) {
+        return this.segmentContainer
+                .forSegment(segmentName, fetchTimeout)
+                .thenComposeAsync(segment -> {
+                    SegmentProperties properties = segment.getInfo();
+                    long compactionOffset = properties.getAttributes().getOrDefault(TableAttributes.COMPACTION_OFFSET, 0L);
```

Comment:
It just occurred to me that this will not work on sorted table segments. It does not have to. 

Let's add a check here (it's an attribute stored in the SegmentProperties) whether this is a sorted table segment. If it is, we should be throwing/returning an UnsupportedOperationException explaining why.